### PR TITLE
feat: remote config

### PR DESCRIPTION
**Pull request by Mercy811**
### Summary

This PR adds the feature remote config.

Note that this PR doesn't merge from the feature branch `feature-remote-config` as it has release commits which introduce conflicts. So I cherry pick all 6 commits from the feature branch `feature-remote-config` without the release commits.

I pushed another commit https://github.com/amplitude/Amplitude-TypeScript/pull/832/commits/637940680f61d060156ccb20fb5c6a9051aa9a40 to upgrade to the latest autocapture plugin 1.x

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

**Review by yuhao900914 — APPROVED** [review]

File: packages/analytics-types/src/element-interactions.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * Default CSS selectors to define which elements on the page to track.
+ * Extend this list to include additional elements to track. For example:
+ * ```
+ * autocapturePlugin({
+ *    cssSelectorAllowlist: [...DEFAULT_CSS_SELECTOR_ALLOWLIST, ".my-class"],
+ * })
+ * ```
+ */
+export const DEFAULT_CSS_SELECTOR_ALLOWLIST = [
```
Comment (Dogfalo) [inline packages/analytics-types/src/element-interactions.ts:14]: Can this be exported from the `plugin-autocapture-browser` instead? That way we don't have to update it in two places.

File: packages/analytics-types/src/element-interactions.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  '[data-amp-default-track]',
+  '.amp-default-track',
+];
+
+/**
+ * Default prefix to allo the plugin to capture data attributes as an event property.
+ */
+export const DEFAULT_DATA_ATTRIBUTE_PREFIX = 'data-amp-track-';
+
+export interface ElementInteractionsOptions {
```
Comment (Dogfalo) [inline packages/analytics-types/src/element-interactions.ts:30]: Same with this interface

Comment (Dogfalo) [reply packages/analytics-types/src/element-interactions.ts:30]: Recently we had to add one additional property inside this called `actionClickAllowlist`, I'm curious if anything needs to change other than this to support it?

File: packages/analytics-types/src/element-interactions.ts (hunk 1 above)
Comment (Mercy811) [reply packages/analytics-types/src/element-interactions.ts:14]: `analytics-types` is a dependency of `plugin-autocapture-browser`. It's a circular dependency if we import these from `plugin-autocapture-browser` in `analytics-types`.

File: packages/analytics-types/src/element-interactions.ts (hunk 2 above)
Comment (Mercy811) [reply packages/analytics-types/src/element-interactions.ts:30]: Once this version is released, we should import these `const` and `interface` from `analytics-types` in `plugin-autocapture-browser`. I copied these from v0.9.2, so we need to all changes after it

File: packages/analytics-types/src/element-interactions.ts (hunk 1 above)
Comment (Dogfalo) [reply packages/analytics-types/src/element-interactions.ts:14]: Sounds good, we can think of some way to resolve this is the next version

**Review by Dogfalo — APPROVED** [review]

File: packages/analytics-types/src/element-interactions.ts (hunk 1 above)
Comment (Mercy811) [reply packages/analytics-types/src/element-interactions.ts:14]: If we keep those in analytics-types, we don’t have circular dependency and this is the pattern other packages, for example page view tracking plugin.

**Pull request merged**